packages/crypto/src/transactions/types/htlc-lock.ts

```diff
@@ -0,0 +1,45 @@
+import ByteBuffer from "bytebuffer";
+import { TransactionTypes } from "../../enums";
+import { ISerializeOptions } from "../../interfaces";
+import { Base58, BigNumber } from "../../utils";
+import * as schemas from "./schemas";
+import { Transaction } from "./transaction";
+
+export class HtlcLockTransaction extends Transaction {
+    public static type: TransactionTypes = TransactionTypes.HtlcLock;
+
+    public static getSchema(): schemas.TransactionSchema {
+        return schemas.htlcLock;
+    }
+
+    public serialize(options?: ISerializeOptions): ByteBuffer {
+        const { data } = this;
+
+        const buffer: ByteBuffer = new ByteBuffer(8 + 32 + 8 + 21, true);
+
+        buffer.writeUint64(+data.amount);
```

Comment:
I think `data.amount` is already packed in the common part of transaction serialization. It is a doublon serializing it here